types/mouse-event-offset/mouse-event-offset-tests.ts

```diff
@@ -0,0 +1 @@
+import offset from 'mouse-event-offset';
```

Comment:
This should become
```suggestion
import offset = require('mouse-event-offset');
```
after the esModuleInterop change. Also, an import is not much of a test 😄

Reply:
I tried to do more, but stubbing out a `MouseEvent` was non-trivial. Any ideas on how to go about this @andrewbranch?

Also, I want to be sure that the default es6 import works. I'm not sure if mixing the `import` keyword with a `require` is valid?

Reply:
> but stubbing out a MouseEvent was non-trivial

You don’t have to stub it:

```ts
declare const mouseEvent: MouseEvent;
declare const touchEvent: TouchEvent;

offset(mouseEvent);
offset(touchEvent);
```

> Also, I want to be sure that the default es6 import works.

The library is not an ES module; it’s a CJS module. A default import will work through many interop layers that exist in the wild, but you can’t assume them when writing universally consumable library code like this. Any user who has `esModuleInterop` enabled will be allowed to use a default import just like you have here. But you aren’t allowed to do it here, because it means you could accidentally use an interop-assuming default import in `index.d.ts` which would cause errors for everyone who uses these types who doesn’t have `esModuleInterop` enabled. Users won’t get the tsconfig file here; it’s only for checking the definitions and tests prior to publishing. So the principle is that you need to run the compilation under the strictest settings, so that you can be sure it will work for everyone.

> I'm not sure if mixing the import keyword with a require is valid?

It’s TypeScript syntax. It compiles to `const whatever = require('...')` in JS.

Reply:
This was super helpful, thank you! Commit incoming.